### PR TITLE
Dịch "optimization-intro_vn" - Phần 1

### LINKED ISSUE #1791: Dịch "optimization-intro_vn" - Phần 1
Phần cần dịch: https://github.com/aivivn/d2l-vn/blame/master/chapter_optimization/optimization-intro_vn.md#L1-L74

### Hướng dẫn
- [Dịch trực tiếp trên website](https://github.com/aivivn/d2l-vn/blob/master/CONTRIBUTING.md).
- Sau khi nhấn vào phần cần dịch ở trên, hãy chọn **Normal view** và làm theo hướng dẫn ở trên để có thể bắt đầu dịch.

### PR DESCRIPTION
**Pull request by trGiang99**
## Pull Request Checklist

**Đánh dấu tick (sau khi "Create Pull Request") vào các mục dưới đây:**

* [x] Pull Request này tương ứng với issue nào? Trả lời số issue sau dấu `#` (không có dấu cách): Close #1791

* [x] Bản dịch này có bám sát **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)** không? Nếu một từ/cụm chưa có trong Bảng thuật ngữ, hãy tạo một Pull Request riêng đề xuất cách dịch từ/cụm đó.

* [x] Format của bản gốc có được giữ nguyên không?
    * Các phần in nghiêng, in đậm.
    * Tiêu đề (số lượng dấu `#` đầu dòng).
    * Bảng biểu, chú thích cho bảng (dòng phía trên mỗi bảng bắt đầu bằng dấu `:`).
    * Dịch các từ trong hình vẽ nếu cần. Các dòng có hình có dạng: `caption`.
    * Dịch các chú thích hình vẽ (thay các cụm `*dịch chú thích ảnh phía trên*` bằng bản dịch tiếng Việt).
    * Không dịch các phần code (nằm giữa hai dấu `````).
    * Copy các công thức toán từ bản gốc (các đoạn có `$`).
    * Giữ các dòng gán nhãn (bắt đầu với `:label:`).
    * Không tự thêm bớt các dòng trắng.
     
* [x] Trong một đoạn văn, mỗi câu văn đã được viết trong một dòng, giữa các dòng không có dòng trắng. Markdown sẽ coi những dòng liền nhau không có dòng trắng là một đoạn văn. Việc này giúp công đoạn review được thuận tiện hơn.

* [ ] Tất cả chữ trong các hình vẽ đã được dịch. Nếu chưa, vui lòng xem thêm **[hướng dẫn hoặc yêu cầu dịch hình](https://github.com/aivivn/d2l-vn/blob/master/img/README.md)**.

* [x] Điền tên của bạn vào mục "Những người thực hiện" ở cuối file.

* [ ] **Sau khi được approve**, điền tên những reviewers mà bạn thấy có nhiều đóng góp cho Pull Request này. Xem danh sách Contributors **[tại đây](https://github.com/aivivn/d2l-vn/blob/master/docs/contributors_info.md)**.

## Với Reviewer

1. Chỉ review nếu Pull Request này đã sẵn sàng cho việc review.

2. Đã đọc kỹ hướng dẫn đóng góp và xem qua **[Bảng thuật ngữ](https://github.com/aivivn/d2l-vn/blob/master/glossary.md)**.

3. **Bạn có thể đề xuất thay đổi cách dịch mỗi dòng trực tiếp như trong hình dưới đây:**
img

### CODE REVIEW

File: chapter_optimization/optimization-intro_vn.md [hunk 1]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
 If we ever need to maximize an objective there is a simple solution: just flip the sign on the objective.
 -->
 
-*dịch đoạn phía trên*
+Trong phần này, ta sẽ thảo luận mối quan hệ giữa tối ưu và học sâu, cũng như những thử thách trong việc áp dụng các thuật toán tối ưu trong học sâu.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:19]:
```suggestion
Trong phần này, ta sẽ thảo luận mối quan hệ giữa tối ưu và học sâu, cũng như những thách thức khi áp dụng các thuật toán tối ưu trong học sâu.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 2]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
+Đối với một bài toán trong học sâu, chúng ta thường định nghĩa hàm mất mát đầu tiên.
+Sau đó ta có thể sử dụng một thuật toán tối ưu nào đó nhằm tối thiểu hoá hàm mất mát mà ta vừa định nghĩa.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:21]:
```suggestion
Sau đó ta có thể sử dụng một thuật toán tối ưu nào đó nhằm cực tiểu hoá hàm mất mát đó.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 3]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
+Trong tối ưu, hàm mất mát thường được xem là hàm mục tiêu của bài toán tối ưu.
+Đa số các hàm tối ưu trong thực tế thường quan tâm đến *tối thiểu hoá*.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:23]:
```suggestion
Theo định nghĩa, đa số các hàm tối ưu thường quan tâm đến *cực tiểu hoá*.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 4]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
+Tuy nhiên nếu ta cần tối đa hoá một mục tiêu nào đó, có một cách khá đơn giản: chỉ cần đổi dấu của mục tiêu.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:24]:
```suggestion
Tuy nhiên, nếu ta cần cực đại hoá một hàm mục tiêu nào đó, có một cách khá đơn giản: chỉ cần đổi dấu hàm mục tiêu.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 5]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
-However, the goal of statistical inference (and thus of deep learning) is to reduce the generalization error. 
+For instance, training error and generalization error generally differ: since the objective function of the optimization algorithm is usually a loss function.
+Based on the training dataset, the goal of optimization is to reduce the training error.
+However, the goal of statistical inference (and thus of deep learning) is to reduce the generalization error.
 To accomplish the latter we need to pay attention to overfitting in addition to using the optimization algorithm to reduce the training error.
 We begin by importing a few libraries with a function to annotate in a figure.
 -->
 
-*dịch đoạn phía trên*
+Mặc dù tối ưu thường được sử dụng để tối thiểu hoá hàm mất mát trong học sâu, về bản chất, mục đích của tối ưu và học sâu là khác nhau.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:44]:
```suggestion
Mặc dù tối ưu thường được sử dụng để cực tiểu hoá hàm mất mát trong học sâu, về bản chất, mục đích của tối ưu và học sâu là khác nhau.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 6]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
+Mối quan tâm của tối ưu chủ yếu nằm ở việc tối thiểu hoá một mục tiêu nào đó, trong khi đối với học sâu là việc tìm kiếm một mô hình phù hợp khi có một lượng dữ liệu có hạn.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:45]:
```suggestion
Mối quan tâm của tối ưu chủ yếu nằm ở việc cực tiểu hoá một mục tiêu nào đó, trong khi đối với học sâu là việc tìm kiếm một mô hình phù hợp với một lượng dữ liệu hữu hạn.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 7]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
+Trong :numref:`sec_model_selection`, ta thảo luận chi tiết về sự khác nhau giữa các mục đích trên.
+Ví dụ, lỗi huấn luyện và lỗi khái quát nhìn chung là khác nhau: do hàm mục tiêu của thuật toán tối ưu thường là hàm mất mát.
+Mục đích của tối ưu là giảm thiểu lỗi huấn luyện dựa theo tập dữ liệu huấn luyện.
+Tuy nhiên, mục đích của suy luận thống kê (*statistical inference*) là giảm thiểu lỗi khái quát.
+Để có thể thực hiện được điều này, ta cần chú ý đến hiện tượng quá khớp khi sử dụng thêm thuật toán tối ưu nhằm giảm thiểu lỗi huấn luyện.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:50]:
```suggestion
Để thực hiện được điều này, ta cần chú ý đến hiện tượng quá khớp bên cạnh việc sử dụng thuật toán tối ưu nhằm giảm thiểu lỗi huấn luyện.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 8]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
+Hãy bắt đầu bằng việc nhập một số thư viện với chức năng tạo chú thích cho ảnh.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:51]:
```suggestion
Hãy bắt đầu bằng việc nhập một số thư viện với hàm để chú thích cho các hình đồ thị dưới đây.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 9]
```diff
@@ -57,7 +69,8 @@ The graph below illustrates the issue in some more detail.
 Since we have only a finite amount of data the minimum of the training error may be at a different location than the minimum of the expected error (or of the test error).
 -->
 
-*dịch đoạn phía trên*
+Đồ thị phía dưới sẽ mô tả chi tiết hơn về vấn đề.
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:72]:
```suggestion
Đồ thị phía dưới mô tả chi tiết hơn về vấn đề nêu trên.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 10]
```diff
@@ -57,7 +69,8 @@ The graph below illustrates the issue in some more detail.
...
+Do ta chỉ có một lượng dữ liệu giới hạn, cực tiểu của lỗi huấn luyện có thể khác so với cực tiểu của lỗi mà ta mong muốn (hay lỗi trên tập kiểm tra).
```
Comment (davidnvq) [inline chapter_optimization/optimization-intro_vn.md:73]:
```suggestion
Do ta chỉ có một lượng dữ liệu giới hạn, cực tiểu của lỗi huấn luyện có thể khác so với cực tiểu kì vọng của lỗi (hay lỗi trên tập kiểm tra).
```

File: chapter_optimization/optimization-intro_vn.md [hunk 11]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
 Although optimization provides a way to minimize the loss function for deep learning, in essence, the goals of optimization and deep learning are fundamentally different.
-The former is primarily concerned with minimizing an objective whereas the latter is concerned with finding a suitable model, given a finite amount of data. 
+The former is primarily concerned with minimizing an objective whereas the latter is concerned with finding a suitable model, given a finite amount of data.
 In :numref:`sec_model_selection`, we discussed the difference between these two goals in detail.
-For instance, training error and generalization error generally differ: since the objective function of the optimization algorithm is usually a loss function 
-based on the training dataset, the goal of optimization is to reduce the training error.
-However, the goal of statistical inference (and thus of deep learning) is to reduce the generalization error. 
+For instance, training error and generalization error generally differ: since the objective function of the optimization algorithm is usually a loss function.
+Based on the training dataset, the goal of optimization is to reduce the training error.
```
Comment (lkhphuc) [inline chapter_optimization/optimization-intro_vn.md:38]: Bạn sửa bản gốc chỗ này sai ý tác giả rồi, nên dịch ra cũng sai theo. Chỗ này vốn là một câu liên tục mà.
Nếu có sửa câu gốc thì mình thấy ngắt câu từ chỗ dấu ":" mới hợp.

File: chapter_optimization/optimization-intro_vn.md [hunk 12]
```diff
@@ -16,27 +16,39 @@ By tradition and convention most optimization algorithms are concerned with *min
...
 To accomplish the latter we need to pay attention to overfitting in addition to using the optimization algorithm to reduce the training error.
 We begin by importing a few libraries with a function to annotate in a figure.
 -->
 
-*dịch đoạn phía trên*
+Mặc dù tối ưu thường được sử dụng để tối thiểu hoá hàm mất mát trong học sâu, về bản chất, mục đích của tối ưu và học sâu là khác nhau.
+Mối quan tâm của tối ưu chủ yếu nằm ở việc tối thiểu hoá một mục tiêu nào đó, trong khi đối với học sâu là việc tìm kiếm một mô hình phù hợp khi có một lượng dữ liệu có hạn.
+Trong :numref:`sec_model_selection`, ta thảo luận chi tiết về sự khác nhau giữa các mục đích trên.
+Ví dụ, lỗi huấn luyện và lỗi khái quát nhìn chung là khác nhau: do hàm mục tiêu của thuật toán tối ưu thường là hàm mất mát.
+Mục đích của tối ưu là giảm thiểu lỗi huấn luyện dựa theo tập dữ liệu huấn luyện.
```
Comment (lkhphuc) [inline chapter_optimization/optimization-intro_vn.md:48]:
```suggestion
Ví dụ, lỗi huấn luyện và lỗi khái quát nhìn chung là khác nhau. 
Do hàm mục tiêu của thuật toán tối ưu thường là hàm mất mát trên tập huấn luyên nên mục đích của tối ưu là giảm thiểu lỗi huấn luyện.
```

File: chapter_optimization/optimization-intro_vn.md [hunk 13]
```diff
@@ -323,7 +336,7 @@ với dấu `@` ở đầu. Ví dụ: @aivivn.
 
 * Đoàn Võ Duy Thanh
 <!-- Phần 1 -->
-* 
+* Đỗ Trường Giang
```
Comment (trGiang99) [inline chapter_optimization/optimization-intro_vn.md:340]:
```suggestion
* Lê Khắc Hồng Phúc
* Nguyễn Văn Quang
```

File: chapter_optimization/optimization-intro_vn.md (hunk 11 above)
Comment (duythanhvn) [reply chapter_optimization/optimization-intro_vn.md:38]: Rất lưu ý với @trGiang99 là tuyệt đối không chỉnh sửa bất kỳ thông tin gì từ bản gốc được để trong tập tin, em chỉ cần dịch theo nội dung ở trên mà thôi nha. Kể cả việc bản tiếng Anh đã có chỉnh lý thì dự án cũng sẽ cập nhật sau, không phải trong giai đoạn này.

Comment (trGiang99) [reply chapter_optimization/optimization-intro_vn.md]: Vầng em sửa lại rồi ạ.
